Goodwe warning fixes

llagendijk opened the pull request:
Fixed the warnings that you asked me to fix in the Goodwe adapter. Had to break deviceID calculation unfortunately.

llagendijk committed:
Fix warning for usage of assingment in if statement in hash function

llagendijk committed:
Fix warning for usage of += in if statement for ChildID.
Fix logic error in the same if statement.
These fixes unfortunately will change the device ID for some people.

gizmocuz commented on hardware/GoodweAPI.cpp:
```diff
@@ -166,11 +166,12 @@ int GoodweAPI::hash(std::string str)
 	{
 		hash = ((hash << 5) + hash) + c;
+		c = str[i++];
 	}
```
I still find this a bit strange way todo this
You know the length of the string ( str.size() )
You could use

while (i<str.size())

or even

for (i=0; i<str.size(); i++)

Both look a bit better

llagendijk replied:
Good comment. I am still too much C-oriented...
Will include this

gizmocuz commented on hardware/GoodweAPI.cpp:
```diff
@@ -442,9 +443,10 @@ void GoodweAPI::ParseDevice(Json::Value device, std::string sStationId, std::str
+	if ((ChildID >= 0) && (ChildID <= 10)) {
+		ChildID = ChildID + 10;
+	}
 
```
Could there be negative child id's ? maybe the hash should return an uint32_t ?

llagendijk committed:
Incorporated  gizmocuz's comments:
Use str.size and a for loop in hash function.
Changed result of hash to uint32_t and leave it to implicit conversions
in the Send functions to convert unsigned to signed.

gizmocuz commented on hardware/GoodweAPI.cpp:
```diff
@@ -168,11 +169,12 @@ int GoodweAPI::hash(std::string str)
 	int i = 0;
```
could you make it size_t instead of int ;)

llagendijk replied:
I should have known this but missed it :-(

llagendijk committed:
Change loop variable type in hash to size_t.

gizmocuz commented:
Thanks !
